### DISCUSSION TITLE
Email-address vs username

### DISCUSSION THREAD

**Discussion by andriusign** [directus/directus#4237] [Feature Requests]
Why email and not username? Perhaps having to create an account with mandatory email address is not a big of a deal until it comes to ensuring privacy.

From the user's point of view: what do I need to register? I need something that I have control over, that would be _username_ and _password_, both of which I control and no one else knows does. I would create an account with this information. This also ensures my privacy, unless I specifically want to put my name and surname in my username.

By having to enter _email_ as a required parameter in order to create an account, I am providing something that I do not have control over (in most cases, unless I run my own email server with own hardware on premises). So theoretically, after registration is complete, my email provider - which in most cases would be a bit tech company - can instantly have access to the entire system, since they control my email and can reset my password anytime without me even knowing.

In case of forgotten password there are also other ways to restore it, etc. security questions. Unless I specifically enter my email address, it should not be used as a main (and only) method to restore my password. 

_email_ as a mandatory field does not prevent spam: fake bots can still register, unless we stop them with Captcha, and troll users can still register with temporary email accounts). It also does not identify the person (in any way that you could rely) so why is it being user as a main identifier to login/register to Directus and not _username_? Perhaps these thoughts sound strange to someone who does not understand anything about encryption or security, but for those who do Directus should be able to provide this extra layer of security.

**Comment by rijkvanzanten** [comment]
Funny that you bring it up. In my recent experience I noticed that more and more services rely on email as primary (and often only) way to register for accounts, even going as far as getting rid of passwords completely in favor of using "magic sign in" links in emails.

**Reply by benhaynes** [reply]
Just to add to this — I suppose for some _very_ secure applications an email could be seen as a security concern, but certainly not for most (80%+) users. And for those hyper-secure edge-cases, it seems very possible to use your own email server to secure things. Also, we don't have public user "Registration" yet, so that's not an issue.

If the issue is privacy, then I think there are things we could do to hide the email address within the App (already possible with permissions).

As of now, an email is a very good unique identifier that mitigates issues of forgotten usernames and security questions. As always, we're down to discuss, but this needs to happen through the lens of our 80/20 rule.

**Reply by sercanvirlan** [reply]
I completely agree with you guys, just an addition for this convo, most mobile applications use phone number as unique identifier with using OTP authentication not a password.

I know this does not fit 80/20 rule but if we think about mobile applications perspective, maybe this is 50/50 need.

**Reply by brunoribei-ro** [reply]
@benhaynes 
I'm wondering if it's possible to have it as an option or custom configuration as I've been working on a project that requires a user to log in with his CPF (which is the Brazilian social security number).

Despite knowing how to add this new field to the user collection, I can't figure out how to use it when calling auth/login endpoint. 

Thank You 👍

**Reply by benhaynes** [reply]
We don't have full Auth Adapters as extensions yet (though I'm down to get these added!), but perhaps you could use/create an SSO provider?

**Comment by rijkvanzanten** [comment]
Heya! Thank you for taking the time to submit this request! 

It has been over **90 days**, and this discussion has not received at least **15 votes** from the community. This means that we don't feel like there's enough community interest to warrant further R&D into this topic at this time. :ice_cube:

This request will now be closed to keep our discussions tidy. Please reach out if you have any questions!

_For more information, see our [Feature Request Process](https://docs.directus.io/contributing/feature-request-process.html)._
